src/com/facebook/buck/apple/project_generator/ProjectGenerator.java

```diff
@@ -2685,7 +2695,14 @@ ProductType bundleToTargetProductType(
             return ProductType.APPLICATION;
         }
       } else if (binaryNode.getType().equals(AppleTestDescription.TYPE)) {
-        return ProductType.UNIT_TEST;
+        @SuppressWarnings("unchecked")
+        TargetNode<AppleTestDescription.Arg> testNode =
+              (TargetNode<AppleTestDescription.Arg>) binaryNode;
+        if (testNode.getConstructorArg().isUiTest()) {
```

Comment:
I couldn't find a typesafe way to express that binaryNode has `AppleTestDescription.Arg`, but after the typecheck this should be true. Suggestions welcome.